spec/rspec/core/configuration_spec.rb

```diff
@@ -1819,6 +1819,20 @@ def exclude?(line)
         expect(group.metadata).to include(:b1_desc => "bar (block 1)", :b2_desc => "bar (block 1) (block 2)")
       end
 
+      it 'supports cascades of derived metadata, but avoids re-running derived metadata blocks that have already been applied' do
```

Comment:
I see how this demonstrates the cascade, but not the re-applied logic...

Reply:
It demonstrates that the blocks are not re-applied by asserting that the values of each of `foo*` is `1`.  The blocks each increment the value, so if any of the `define_derived_metadata` blocks are run multiple times, then the valule would be something greater than 1 and the expectation below would fail.

How can I make this more clear?

Reply:
I wouldn't expect in this scenario the blocks to be run more than once though, but this would demonstrate the issue more clearly:

```
c.define_derived_metadata(:foo1) { |m| m[:foo2] = (m[:foo2] || 0) + 1 }
c.define_derived_metadata(:foo2) { |m| m[:foo3] = (m[:foo3] || 0) + 1 }
c.define_derived_metadata(:foo3) { |m| m[:foo1] = (m[:foo1] || 0) + 1 }

group = RSpec.describe("bar", :foo1 => 1)
expect(group.metadata).to include(:foo1 => 2, :foo2 => 1, :foo3 => 1)
```

Reply:
> I wouldn't expect in this scenario the blocks to be run more than once though

They would run  more  than once if it wasn't for the `unless skip block` bit on the `block.call(metadata) unless skip_block` line.

Anyhow, I've changed to:

``` ruby
        RSpec.configure do |c|
          c.define_derived_metadata(:foo1) { |m| m[:foo2] = (m[:foo2] || 0) + 1 }
          c.define_derived_metadata(:foo2) { |m| m[:foo3] = (m[:foo3] || 0) + 1 }
          c.define_derived_metadata(:foo3) { |m| m[:foo1] += 1 }
        end

        group = RSpec.describe("bar", :foo1 => 0)
        expect(group.metadata).to include(:foo1 => 1, :foo2 => 1, :foo3 => 1)
```

I think it's a bit easier to see that each block runs only once if the values are all `1` (if one value is `2` it makes it seem like a block ran twice I think), so we can start `foo1` at 0 to achieve this.